src/lib/scheduler/no_scheduler.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "abstract_scheduler.hpp"
+#include "abstract_task.hpp"
+
+namespace opossum {
+
+class NoScheduler : public AbstractScheduler {
```

Comment:
Naming: `ImmediateExecutionScheduler`? I'll admit that it is bulky, maybe you have a better idea?